homeassistant/components/ps4/media_player.py

```diff
@@ -0,0 +1,390 @@
+    config = config_entry
+
+    def add_devices(devices, update_before_add=False):
+        """Sync version of async add devices."""
+        hass.add_job(async_add_devices, devices, update_before_add)
+
+    await hass.async_add_executor_job(
+        _setup_platform, hass, config,
+        add_devices, None)
+
+    async def async_service_handle(hass):
+        """Handle for services."""
+        def service_command(call):
+            entity_ids = call.data.get(ATTR_ENTITY_ID)
+            command = call.data.get(ATTR_COMMAND)
+            for device in hass.data[PLATFORM].devices:
+                device = device
+                if device.entity_id in entity_ids:
+                    device.send_command(command)
+
+        hass.services.async_register(
+            PLATFORM, SERVICE_COMMAND, service_command,
+            schema=PS4_COMMAND_SCHEMA)
+
+    await async_service_handle(hass)
+
+
+def _setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up PS4 Platform."""
+    import pyps4_homeassistant as pyps4
+    hass.data[PLATFORM] = PS4Data()
+    games_file = hass.config.path(GAMES_FILE)
+    creds = config.data[CONF_TOKEN]
+    for device in config.data['devices']:
+        host = device.get(CONF_HOST)
+        region = device.get(CONF_REGION)
+        name = device.get(CONF_NAME)
+        ps4 = pyps4.Ps4(host, creds)
+        add_devices([
+            PS4Device(name, config, host, region, ps4, games_file)], True)
+
+
+class PS4Data():
+    """Init Data Class."""
+
+    def __init__(self):
+        """Init Class."""
+        self.devices = []
+
+
+class PS4Device(MediaPlayerDevice):
+    """Representation of a PS4."""
+
+    def __init__(self, name, config, host, region, ps4, games_file):
+        """Initialize the ps4 device."""
+        self._config = config
+        self._ps4 = ps4
+        self._host = host
+        self._name = name
+        self._region = region
+        self._state = STATE_UNKNOWN
+        self._games_filename = games_file
+        self._media_content_id = None
+        self._media_title = None
+        self._media_image = None
+        self._source = None
+        self._games = self.load_games()
+        self._source_list = list(sorted(self._games.values()))
+        self._retry = 0
+        self._info = None
+        self._unique_id = None
+
+    async def async_added_to_hass(self):
+        """Subscribe PS4 events."""
+        self.hass.data[PLATFORM].devices.append(self)
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
+        """Retrieve the latest data."""
+        try:
+            status = self._ps4.get_status()
+            if self._info is None:
+                self.get_device_info(self._config, status)
+        except socket.timeout:
+            status = None
+        if status is not None:
+            self._retry = 0
+            if status.get('status') == 'Ok':
+                titleid = status.get('running-app-titleid')
```

Comment:
Please separate words with underscore in variable names. Let's call this `title_id`.